redaxo/src/core/lib/view.php

```diff
@@ -320,7 +320,10 @@ public static function title($head, $subtitle = null)
         $fragment->setVar('subtitle', $subtitle, false);
         $return = $fragment->parse('core/page/header.php');
 
-        $return .= rex_extension::registerPoint(new rex_extension_point('PAGE_TITLE_SHOWN', ''));
+        /** @psalm-taint-escape html */ // we expect safe content from the extensions
```

Comment:
ggf. macht es sogar sinn `/** @psalm-taint-escape html */` direkt an der EP Klasse zu notieren, da wir immer valides/escapte inhalte von den EPs erwarten?

Reply:
wäre sehr praktisch, da es an sehr vielen stellen so ist.

Gibt aber auch Gegenbeispiele, eins habe ich gerade gefunden: https://github.com/redaxo/redaxo/blob/master/redaxo/src/core/lib/util/i18n.php#L185
Dort wird kein sicherer Content erwartet, und es muss daher später escaped werden (und wird es auch).

Aber vielleicht ist es trotzdem ok, es global so zu setzen, und bei den paar wenigen Stellen, wo es anders ist, quasi auf die weitere taint analyse zu verzichten?

Reply:
hmm für mich wiederspricht sich das nicht. die extension in diesem Beispiel liefert einen plain string (sie rendert kein html in dessen context sie escapen müsste bzw. auch kein sql query in welchem sie escapen müsste)

aber vllt macht es auch sinn wenn wir subklassen für die inhaltstypen anlegen, d.h. ein `rex_extension_point_html` für einen EP der ein html subject handelt und ensprechend html-context tain-escaping hat, analog für andere contexte?

Reply:
Habe es jetzt mal global gemacht. Ob das gut ist, bin ich mir weiterhin unsicher.
Aber wäre für mich ok, erstmal so zu machen.

Neue Unterklassen dafür fände ich nicht so richtig cool.
Wir haben ja auch EPs mit Arrays als Subjekt. Theoretisch ist da auch denkbar, dass ein Value für HTML-Content ist, und ein anderes nicht.

Reply:
Nochmal drüber nachgedacht. Wir könnten das hier so machen, wie ich es jetzt habe. Also den Output von `registerPoint` grundsätzlich als sicher markieren.
Und dann könnten wir eventuell mit einem Plugin `rex_extension::register` als sink definieren, mit statischer Liste, für die EPs, wo das der Fall ist. So hätten wir dann wieder die nötigen Checks für den Input in die Extension rein, da wo nötig.
Denn psalm wird ja selbst sowieso nicht die Verknüpfung ziehen können zwischen EP und den zugehörigen Extensions.

Reply:
Wobei es nicht nur für den Input in die Extensions greifen muss, sondern ja doch auch für den Input in das initiale Subjekt im Extension Point.

Wie auch immer: Stand jetzt wäre ich für hier erstmal mergen. Und dann bei Gelegenheit noch mal schauen, ob man es nochmal optimiert, und Checks, die wir hier zu global deaktiviert haben, wieder selektiv aktiviert bekommt.

Reply:
Alternativ könnte man eine selarate baseline für taint verwenden

https://psalm.dev/docs/security_analysis/#using-baseline-with-taint-analysis

Reply:
ja, baseline ist wahrscheinlich so oder so sinnvoll, gerade weil unser type-plugin zum beispiel noch nicht greift. und auch die conditional escapes lassen sich wohl doch noch nicht wirklich lösen.
daher gibt es vermutlich viele stellen, wo man besser auf die richtige lösung wartet, als jetzt mit workarounds mit  vielen eigentlich unnötigen annotationen zu arbeiten. daher gerne baseline.

unabhängig davon könnte man das hier für die eps trotzdem so reinnehmen. im moment vermute ich, dass meine änderung hier auch teil einer mehr korrekten lösung für die EPs wäre.
Aber bin auch einverstanden, wenn wir es doch lieber weglassen, und die fälle auch in die baseline kommen.

Reply:
baseline ist jetzt da.. man könnte jetzt überlegen wie selektiv wir hier weiter vorgehen wollen

Reply:
Nochmal weiter überlegt, und jetzt specialize gesetzt.
Damit ist jetzt erreicht, dass bei `rex_extension::registerPoint` das korrekte initiale Subjekt betrachtet wird, und nicht initiale Subjekte von anderen EPs.
Was aber fehlt: Die Verknüpfung der Extensions mit dem Extension Point. Da denke ich weiterhin, das lässt sich nur über ein Plugin lösen, wenn überhaupt. Die sind ja an völlig anderen Codestellen.

Ich denke, man könnte das hier so aber ruhig erstmal mergen. Die Kreuz-und-quer-Verknüpfungen vorher zwischen den verschiedenen EPs machte keinen Sinn.